dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -271,13 +271,21 @@ private static DependencyGraph levelOrder(
     while (!queue.isEmpty()) {
       LevelOrderQueueItem item = queue.poll();
       DependencyNode dependencyNode = item.dependencyNode;
-      DependencyPath forPath = new DependencyPath();
+      DependencyPath path = new DependencyPath();
       Stack<DependencyNode> parentNodes = item.parentNodes;
-      parentNodes.forEach(parentNode -> forPath.add(parentNode.getArtifact()));
+      parentNodes.forEach(
+          parentNode ->
+              path.add(
+                  parentNode.getArtifact(),
```

Comment:
Should `DependencyPath.add()` take a `DependencyNode` and decompose it itself?

Reply:
That's a nice suggestion. But I would keep the three parameters. The idea of taking `DependencyNode` would make this line simpler, but other places, such as DependencyPathTest, would become complex like this:

```
add(new DefaultDependencyNode(new Dependency(foo, "compile", false)));
```

Reply:
You could have an overload.